components/EndingScene.ts

```diff
@@ -0,0 +1,146 @@
+import { ThirdwebSDK } from "@thirdweb-dev/sdk";
+import { ethers } from "ethers";
+import Phaser from "phaser";
+
+const CHAIN_ID = 80001;
+
+export default class EndingScene extends Phaser.Scene {
+  score = 0;
+  recordSeconds = 0;
+  username = "";
+  nftTitle: Phaser.GameObjects.Text | undefined;
+
+  constructor() {
+    super({ key: "ending" });
+  }
+
+  init(data: any) {
+    this.score = data.score;
+    this.recordSeconds = data.recordTime / 1000;
+  }
+
+  preload() {
+    this.load.html("mintform", "assets/form.html");
+  }
+
+  create() {
+    this.add.image(400, 300, "pic");
+
+    this.nftTitle = this.add.text(10, 10, "Submit your nickname to mint", {
+      color: "white",
+      fontFamily: "Arial",
+      fontSize: "32px ",
+    });
+
+    var element = this.add.dom(400, 300).createFromCache("mintform");
+
+    element.setPerspective(800);
+
+    element.addListener("click");
+
+    element.on("click", (event: any) => {
+      if (event.target.name === "submitButton") {
+        //  Turn off the click events
+        element.removeListener("click");
+
+        var inputUsername: any = element.getChildByName("username");
+
+        //  Have they entered anything?
+        if (inputUsername.value !== "") {
+          //  Populate the text with whatever they typed in as the username!
+          if (this.nftTitle) {
+            this.nftTitle.setText(
+              "Congrats " +
+                inputUsername.value +
+                " for scoring " +
+                this.score +
+                " in " +
+                this.recordSeconds +
+                " seconds. "
+            );
+
+            this.username = inputUsername.value;
+            this.mintWithSignature();
+          }
+        } else {
+          //  Flash the prompt
+          alert("Write your nickname");
+        }
+      }
+    });
+  }
+
+  // This function calls a Next JS API route that mints an NFT with signature-based minting.
```

Comment:
Can you ask @joaquim-verges if there is a better way to do this with thirdweb rather than ethers like this?

There must be a way to do this with TS alone and not React but not sure the syntax myself

Reply:
switched ethers to the new @thirdweb-dev/wallets lib